protocol/parser/parser.go

```diff
@@ -122,105 +123,120 @@ func parseInputFromParamsWithGenericParsers(rpcInput RPCInput, genericParsers []
+			if err != nil {
+				utils.LavaFormatError("Failed parsing default value, setting to NOT_APPLICABLE", err,
+					utils.LogAttr("default_value", defaultValue),
 				)
-				parsedBlock, err = rpcInput.ParseBlock(blockParser.DefaultValue)
-				if err != nil {
-					utils.LavaFormatError("Failed parsing default value, setting to NOT_APPLICABLE", err,
-						utils.LogAttr("default_value", blockParser.DefaultValue),
-					)
-					return spectypes.NOT_APPLICABLE
-				}
-			} else {
-				return spectypes.NOT_APPLICABLE
+				parsedBlock = spectypes.NOT_APPLICABLE
 			}
+		} else {
+			parsedBlock = spectypes.NOT_APPLICABLE
 		}
-		return parsedBlock
-	}()
-
-	return parsedBlockInfo
+	}
+	parsedInput.SetBlock(parsedBlock)
 }
 
-// This returns the parsed response without decoding
-func ParseFromReply(rpcInput RPCInput, blockParser spectypes.BlockParser) (string, error) {
-	result, err := parse(rpcInput, blockParser, PARSE_RESULT)
+func parseInputWithLegacyBlockParser(rpcInput RPCInput, blockParser spectypes.BlockParser, source int) (string, error) {
+	result, err := legacyParse(rpcInput, blockParser, source)
 	if err != nil || result == nil {
-		utils.LavaFormatDebug("ParseBlockFromParams - parse failed",
+		return "", utils.LavaFormatDebug("blockParsing - parse failed",
 			utils.LogAttr("error", err),
 			utils.LogAttr("result", result),
 			utils.LogAttr("blockParser", blockParser),
 			utils.LogAttr("rpcInput", rpcInput),
 		)
-		return "", err
 	}
 
-	response, ok := result[spectypes.DEFAULT_PARSED_RESULT_INDEX].(string)
+	resString, ok := result[spectypes.DEFAULT_PARSED_RESULT_INDEX].(string)
 	if !ok {
-		return "", utils.LavaFormatError("Failed to Convert blockData[spectypes.DEFAULT_PARSED_RESULT_INDEX].(string)", nil, utils.Attribute{Key: "blockData", Value: response[spectypes.DEFAULT_PARSED_RESULT_INDEX]})
+		return "", utils.LavaFormatDebug("blockParsing - result[0].(string) - type assertion failed", utils.LogAttr("result[0]", result[0]))
 	}
 
-	if strings.Contains(response, "\"") {
-		responseUnquoted, err := strconv.Unquote(response)
-		if err != nil {
-			return response, nil
-		}
-		return responseUnquoted, nil
+	return resString, nil
+}
+
+func parseBlock(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser, source int) *ParsedInput {
+	parsedBlockInfo, parsedSuccessfully := parseInputWithGenericParsers(rpcInput, genericParsers)
+	if parsedSuccessfully {
+		parsedBlockInfo.parsedDataRaw = unquoteString(parsedBlockInfo.parsedDataRaw)
+		return parsedBlockInfo
+	}
+	if parsedBlockInfo == nil {
+		parsedBlockInfo = NewParsedInput()
 	}
 
-	return response, nil
+	parsedRawBlock, _ := parseInputWithLegacyBlockParser(rpcInput, blockParser, source)
+	parsedBlockInfo.parsedDataRaw = unquoteString(parsedRawBlock)
+	return parsedBlockInfo
+}
+
+func ParseRawBlockFromParams(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser) *ParsedInput {
```

Comment:
comments, how is that different from the other parse block, why do we need the parseRawBlock? didn't see it being used

Reply:
Was originally used by the `Verify` function of the chain fetcher, but then changed it again and forgot to remove.
Removed in 3e69eaa719b0786f13c04d6362a0d566df004505